app/middlewares/validators/users.js

```diff
@@ -0,0 +1,28 @@
+const UserModel = require('../../models/user');
+const errors = require('../../errors');
+
+exports.checkExistingEmail = async email => {
+  const user = await UserModel.findOne({ where: { email } });
+
+  if (user) throw errors.badRequestError('User already exists');
```

Comment:
badRequestError and 400 is too generic for this, since we know the actual error. Create a more specific function and check the 409 code, I think it's better suited for this.